Dockerfile

```diff
@@ -0,0 +1,25 @@
+ARG PYTHON_VERSION=3.12
+FROM python:${PYTHON_VERSION}-slim AS base
+LABEL maintainer="vlados.soc@gmail.com"
+
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PYTHONUNBUFFERED=1
+
+WORKDIR /app
+
+COPY requirements.txt requirements.txt
+RUN pip install --no-cache-dir -r requirements.txt
+
+COPY . .

```

Comment:
Consider using a more specific COPY command to avoid copying unnecessary files into the Docker image. For example, you can use `.dockerignore` to exclude files that are not needed in the image.